src/hotspot/cpu/riscv/nativeInst_riscv.hpp

```diff
@@ -351,26 +377,33 @@ inline NativeCall* nativeCall_before(address return_address) {
 class NativeMovConstReg: public NativeInstruction {
  public:
   enum RISCV_specific_constants {
-    movptr_instruction_size             =    6 * NativeInstruction::instruction_size, // lui, addi, slli, addi, slli, addi.  See movptr().
-    load_pc_relative_instruction_size   =    2 * NativeInstruction::instruction_size, // auipc, ld
-    instruction_offset                  =    0,
-    displacement_offset                 =    0
+    movptr1_instruction_size            =    6 * NativeInstruction::instruction_size, // lui, addi, slli, addi, slli, addi.  See movptr1().
+    movptr2_instruction_size            =    5 * NativeInstruction::instruction_size, // lui, lui, slli, add, addi.  See movptr2().
+    load_pc_relative_instruction_size   =    2 * NativeInstruction::instruction_size  // auipc, ld
   };
 
-  address instruction_address() const       { return addr_at(instruction_offset); }
+  address instruction_address() const       { return addr_at(0); }
   address next_instruction_address() const  {
     // if the instruction at 5 * instruction_size is addi,
     // it means a lui + addi + slli + addi + slli + addi instruction sequence,
     // and the next instruction address should be addr_at(6 * instruction_size).
     // However, when the instruction at 5 * instruction_size isn't addi,
     // the next instruction address should be addr_at(5 * instruction_size)
-    if (nativeInstruction_at(instruction_address())->is_movptr()) {
-      if (is_addi_at(addr_at(movptr_instruction_size - NativeInstruction::instruction_size))) {
+    if (is_movptr1_at(instruction_address())) {
+      if (is_addi_at(addr_at(movptr1_instruction_size - NativeInstruction::instruction_size))) {
+        // Assume: lui, addi, slli, addi, slli, addi
+        return addr_at(movptr1_instruction_size);
+      } else {
+        // Assume: lui, addi, slli, addi, slli
+        return addr_at(movptr1_instruction_size - NativeInstruction::instruction_size);
+      }
+    } else if (is_movptr2_at(instruction_address())) {
+      if (is_addi_at(addr_at(movptr2_instruction_size - NativeInstruction::instruction_size))) {
         // Assume: lui, addi, slli, addi, slli, addi
```

Comment:
If it's a `movptr2` here, the `Assume: lui, addi, slli, addi, slli, addi` comment seems wrong. Same at https://github.com/openjdk/jdk/pull/19246/files#diff-cadab323d5b577bec017d8ed262bff9d2318e38c0a62afe567050e86ff62cbb9R405